mod/lostpass.php

```diff
@@ -65,109 +65,110 @@ function lostpass_post(App $a) {
 		The login details are as follows:
 
 		Site Location:	%2$s
-		Login Name:	%3$s'));
-
-	$preamble = sprintf($preamble, $username, $sitename);
-	$body = sprintf($body, $resetlink, System::baseUrl(), $email);
+		Login Name:	%3$s', $resetlink, System::baseUrl(), $user['email']));
 
 	notification([
-		'type' => SYSTEM_EMAIL,
-		'to_email' => $email,
-		'subject'=> sprintf( t('Password reset requested at %s'),$sitename),
-		'preamble'=> $preamble,
-		'body' => $body]);
+		'type'     => SYSTEM_EMAIL,
+		'to_email' => $user['email'],
+		'subject'  => t('Password reset requested at %s', $sitename),
+		'preamble' => $preamble,
+		'body'     => $body
+	]);
 
 	goaway(System::baseUrl());
-
 }
 
+function lostpass_content(App $a)
+{
+	$o = '';
+	if ($a->argc > 1) {
+		$pwdreset_token = $a->argv[1];
 
-function lostpass_content(App $a) {
+		$user = dba::selectFirst('user', ['uid', 'username', 'email', 'pwdreset_time'], ['pwdreset' => $pwdreset_token]);
+		if (!DBM::is_result($user)) {
+			notice(t("Request could not be verified. \x28You may have previously submitted it.\x29 Password reset failed."));
 
+			return lostpass_form();
+		}
 
-	if(x($_GET,'verify')) {
-		$verify = $_GET['verify'];
-		$hash = hash('whirlpool', $verify);
+		// Password reset requests expire in 20 minutes
```

Comment:
20 Minutes could be too short if servers are using greylisting. I would increase to one hour.

Reply:
What's greylisting?

Reply:
See here: https://en.wikipedia.org/wiki/Greylisting

It's a method to fight against spam.

Reply:
Geez, that doesn't help us.

Reply:
Yeah, that's funny when because of this your password reminder arrives when its too late.